datetime/datetime.go

```diff
@@ -94,6 +94,99 @@ func NewDatetime(t time.Time) (*Datetime, error) {
 	return dt, nil
 }
 
+func intervalFromDatetime(dtime *Datetime) (ival Interval) {
+	ival.Year = int64(dtime.time.Year())
+	ival.Month = int64(dtime.time.Month())
+	ival.Day = int64(dtime.time.Day())
+	ival.Hour = int64(dtime.time.Hour())
+	ival.Min = int64(dtime.time.Minute())
+	ival.Sec = int64(dtime.time.Second())
+	ival.Nsec = int64(dtime.time.Nanosecond())
+	ival.Adjust = NoneAdjust
+
+	return ival
+}
+
+func daysInMonth(year int64, month int64) int64 {
+	if month == 12 {
+		year++
+		month = 1
+	} else {
+		month += 1
+	}
+	return int64(time.Date(int(year), time.Month(month), 0, 0, 0, 0, 0, time.UTC).Day())
+}
+
+// C imlementation:
+// https://github.com/tarantool/c-dt/blob/cec6acebb54d9e73ea0b99c63898732abd7683a6/dt_arithmetic.c#L74-L98
+func addMonth(ival *Interval, delta int64, adjust Adjust) {
+	oldYear := ival.Year
+	oldMonth := ival.Month
+
+	ival.Month += delta
+	if ival.Month < 1 || ival.Month > 12 {
+		ival.Year += ival.Month / 12
+		ival.Month %= 12
+		if ival.Month < 1 {
+			ival.Year--
+			ival.Month += 12
+		}
+	}
+	if adjust == ExcessAdjust || ival.Day < 28 {
+		return
+	}
+
+	dim := daysInMonth(ival.Year, ival.Month)
+	if ival.Day > dim ||
+		(adjust == LastAdjust && ival.Day == daysInMonth(oldYear, oldMonth)) {
```

Comment:
Formatting in L141-142 is kinda confusing.

Reply:
I've merge it into a one line. It's not too long.